consul/pool.go

```diff
@@ -262,9 +273,16 @@ func (p *ConnPool) acquire(dc string, addr net.Addr, version int) (*Conn, error)
 }
 
 // getNewConn is used to return a new connection
-func (p *ConnPool) getNewConn(dc string, addr net.Addr, version int) (*Conn, error) {
+func (p *ConnPool) getNewConn(dc string, addr net.Addr, version int, deadline time.Time) (*Conn, error) {
+	var timeout time.Duration
+	if deadline.IsZero() {
+		timeout = defaultConnDialTimeout
+	} else {
+		timeout = deadline.Sub(time.Now())
```

Comment:
If this is negative (not likely in actual use) do you just want to bail immediately?